src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -37,6 +37,99 @@ public AnalyzeCommandBaseTests(ITestOutputHelper output)
             Output.WriteLine($"The seed that will be used is: {TestRule.s_seed}");
         }
 
+        [Fact]
+        public void AnalyzeCommand_AbsoluteFileURIWithPathsContainsIllegalCharacters()
+        {
+            var logger = new TestMessageLogger();
+            var command = new TestMultithreadedAnalyzeCommand();
+
+            // No content is set for the following uri, so when its length or content is retrieved,
+            // the file path format will be evaluated.
+            var uri1 = new Uri("C:\\test" + Path.GetInvalidPathChars()[0] + "ABC.md", UriKind.Absolute);
+            var target1 = new EnumeratedArtifact(FileSystem.Instance) { Uri = uri1 };
+
+            // A normal scan target.
+            var uri2 = new Uri("C:\\testABC.md", UriKind.Absolute);
+            var target2 = new EnumeratedArtifact(FileSystem.Instance) { Uri = uri2, Contents = "foo foo" };
+
+            var options = new TestAnalyzeOptions
+            {
+                PluginFilePaths = new[] { typeof(TestRule).Assembly.FullName },
+            };
+
+            var context = new TestAnalysisContext
+            {
+                TargetsProvider = new ArtifactProvider(new[] { target1, target2 }),
+                Logger = logger,
+            };
+
+            int result = command.Run(options: null, ref context);
+            context.ValidateCommandExecution(result);
+            context.RuntimeErrors.Should().Be(RuntimeConditions.OneOrMoreFilesSkipped);
+        }
+
+        [Fact]
+        public void AnalyzeCommand_AbsoluteFileURIWithPercentInFile()
+        {
+            var logger = new TestMessageLogger();
+            var command = new TestMultithreadedAnalyzeCommand();
+
+            string filePath = Path.Combine(Path.GetTempPath(), "New%2DYear.md");
+            File.WriteAllText(filePath, $"{Guid.NewGuid}");
+
+            var uri = new Uri(filePath, UriKind.Absolute);
+
+            var target = new EnumeratedArtifact(FileSystem.Instance) { Uri = uri };
+
+            var options = new TestAnalyzeOptions
+            {
+                PluginFilePaths = new[] { typeof(TestRule).Assembly.FullName },
+            };
+
+            var context = new TestAnalysisContext
+            {
+                TargetsProvider = new ArtifactProvider(new[] { target }),
+                Logger = logger,
+            };
+
+            int result = command.Run(options: null, ref context);
+            context.ValidateCommandExecution(result);
+        }
+
+        [Fact]
+        public void AnalyzeCommand_AbsoluteFileURIWithPercentInDirectory()
+        {
+            var logger = new TestMessageLogger();
+            var command = new TestMultithreadedAnalyzeCommand();
+
+            string dirPath = Path.Combine(Path.GetTempPath(), "New%2DYearDir");
+            if (!Directory.Exists(dirPath))
+            {
+                Directory.CreateDirectory(dirPath);
```

Comment:
There is no big difference between this test case and the one above. IMO, you can combine them into one, that is both the directory, and the filename contains encoded characters.

Reply:
Fixed.  Thanks!